### PR TITLE
UCT/IB/MLX5: Fix TX_INLINE_RESP=0 behavior and add tests for it

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Make sure TX_INLINE_RESP=0 will set get.min_zcopy=1

### CODE REVIEW

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 1]
```diff
@@ -962,6 +962,28 @@ void uct_rc_mlx5_iface_common_dm_cleanup(uct_rc_mlx5_iface_common_t *iface)
...
+#if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
+    uct_ib_dir_t dir;
+
+    dv_attr->comp_mask |= MLX5DV_QP_INIT_ATTR_MASK_QP_CREATE_FLAGS;
+    ucs_for_each_bit(dir, scat2cqe_dir_mask) {
+        if (iface->super.super.config.max_inl_cqe[UCT_IB_DIR_TX] == 0) {
+            dv_attr->create_flags = MLX5DV_QP_CREATE_DISABLE_SCATTER_TO_CQE;
+            return;
+        }
+    }
```
Comment (hoopoepg) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:979]: I didn't get this loop. is it just evaluation for ```scat2cqe_dir_mask != 0```? variable dir is not used inside of loop, and in the worst case loop will be repeated few times with same action

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 2]
```diff
@@ -962,6 +962,29 @@ void uct_rc_mlx5_iface_common_dm_cleanup(uct_rc_mlx5_iface_common_t *iface)
...
+    dv_attr->comp_mask |= MLX5DV_QP_INIT_ATTR_MASK_QP_CREATE_FLAGS;
+    ucs_for_each_bit(dir, scat2cqe_dir_mask) {
+        ucs_assert(dir < UCT_IB_DIR_NUM);
+        if (iface->super.super.config.max_inl_cqe[dir] == 0) {
+            dv_attr->create_flags = MLX5DV_QP_CREATE_DISABLE_SCATTER_TO_CQE;
+            return;
+        }
+    }
+
+    /* we allow scatter2cqe only of all directions we use support it */
```
Comment (brminich) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:982]: `of`->`if`

**Comment by yosefe** [comment]
@Artemy-Mellanox can you pls review?

File: src/uct/ib/rc/accel/rc_mlx5_common.c [hunk 3]
```diff
@@ -962,6 +962,29 @@ void uct_rc_mlx5_iface_common_dm_cleanup(uct_rc_mlx5_iface_common_t *iface)
...
+                                       unsigned scat2cqe_dir_mask)
+{
+#if HAVE_DECL_MLX5DV_QP_CREATE_ALLOW_SCATTER_TO_CQE
+    unsigned dir;
+
+    dv_attr->comp_mask |= MLX5DV_QP_INIT_ATTR_MASK_QP_CREATE_FLAGS;
+    ucs_for_each_bit(dir, scat2cqe_dir_mask) {
+        ucs_assert(dir < UCT_IB_DIR_NUM);
+        if (iface->super.super.config.max_inl_cqe[dir] == 0) {
+            dv_attr->create_flags = MLX5DV_QP_CREATE_DISABLE_SCATTER_TO_CQE;
```
Comment (Artemy-Mellanox) [inline src/uct/ib/rc/accel/rc_mlx5_common.c:977]: AFAIU MLX5DV_QP_CREATE_DISABLE_SCATTER_TO_CQE we need to add only if dir == TX, return regardless

File: test/gtest/uct/ib/test_ib_xfer.cc [hunk 4]
```diff
@@ -26,6 +26,38 @@ UCS_TEST_SKIP_COND_P(uct_p2p_rma_test_inlresp, get_bcopy_inlresp64,
...
+    test_xfer_multi(static_cast<send_func_t>(&uct_p2p_rma_test::get_zcopy),
+                    sender().iface_attr().cap.get.min_zcopy,
+                    sender().iface_attr().cap.get.max_zcopy,
+                    TEST_UCT_FLAG_RECV_ZCOPY);
+}
+
+#if HAVE_DEVX
+/* test mlx5dv_create_qp() */
+UCS_TEST_SKIP_COND_P(uct_p2p_rma_test_inlresp, get_zcopy_inlresp0_devx_no,
+                     !check_caps(UCT_IFACE_FLAG_GET_BCOPY),
```
Comment (Artemy-Mellanox) [inline test/gtest/uct/ib/test_ib_xfer.cc:42]: BCOPY or ZCOPY?

File: src/uct/ib/rc/accel/rc_mlx5_common.c (hunk 3 above)
Comment (yosefe) [reply src/uct/ib/rc/accel/rc_mlx5_common.c:977]: doesn't it affect both directions?
e.g if max_inl_cqe[RX]==0, how can we disable scatter2cqe for receiver?

Comment (Artemy-Mellanox) [reply src/uct/ib/rc/accel/rc_mlx5_common.c:977]: isn't it default, i.e. nothing to pass?

**Comment by yosefe** [comment]
@Artemy-Mellanox can you pls review again?

**Pull request merged**